addons/survey/static/src/js/survey_timer.js

```diff
@@ -1,45 +1,71 @@
 odoo.define('survey.timer', function (require) {
 'use strict';
 
-require('web.dom_ready');
+var publicWidget = require('web.public.widget');
 
-if (!$('.js_survey_timer').length) {
-    return Promise.reject("DOM doesn't contain '.js_survey_timer'");
-}
+var interval = null;
+var $timer = null;
+var countDownDate = null;
 
-var $parent = $('.js_survey_timer');
-var timeLimitMinutes = parseInt($parent.data('time_limit_minutes'));
+publicWidget.registry.SurveyTimerWidget = publicWidget.Widget.extend({
+    template: false,
+    events: {},
 
-if (timeLimitMinutes <= 0) {
-    return Promise.reject("Timer is not positive");
-}
+    //--------------------------------------------------------------------------
+    // Widget
+    //--------------------------------------------------------------------------
 
-var countDownDate = moment.utc($parent.data('timer')).add(timeLimitMinutes, 'minutes');
+    /**
+    * @override
+    */
+    start: function () {
+        // Validation
+        if (!$('.o_survey_timer').length) {
+            return Promise.reject("DOM doesn't contain '.o_survey_timer'");
+        }
 
-if (countDownDate.diff(moment.utc(), 'seconds') < 0) {
-    return Promise.reject("Timer is already finished");
-}
+        var $parent = $('.o_survey_timer');
+        var timeLimitMinutes = parseInt($parent.data('time_limit_minutes'));
+        if (timeLimitMinutes <= 0) {
+            return Promise.reject("Timer is not positive");
+        }
 
-var formatTime = function (time) {
-    return time > 9 ? time : '0' + time;
-};
+        countDownDate = moment.utc($parent.data('timer')).add(timeLimitMinutes, 'minutes');
+        if (countDownDate.diff(moment.utc(), 'seconds') < 0) {
+            return Promise.reject("Timer is already finished");
+        }
 
-var $timer = $parent.find('.timer');
-var interval = null;
-var updateTimer = function () {
-    var timeLeft = countDownDate.diff(moment.utc(), 'seconds');
-
-    if (timeLeft >= 0) {
-        var timeLeftMinutes = parseInt(timeLeft / 60);
-        var timeLeftSeconds = timeLeft - (timeLeftMinutes * 60);
-        $timer.html(formatTime(timeLeftMinutes) + ':' + formatTime(timeLeftSeconds));
-    } else {
-        clearInterval(interval);
-        $('.js_surveyform').submit();
-    }
-};
-
-updateTimer();
-interval = setInterval(updateTimer, 1000);
+        // Run timer
+        $timer = $parent.find('.timer');
+        this._updateTimer(this);
+        interval = setInterval(this._updateTimer, 1000, this);
```

Comment:
Try this._updateTimer.bind(this) to use the `this` context in the method instead of passing the widget instance.